app/models/note.rb

```diff
@@ -11,6 +11,23 @@
 #  updated_at :datetime         not null
 #
 class Note < ApplicationRecord
+  scope :by_filter, lambda { |filters|
+    notes = all
+    compacted_filter = filters.compact
+    compacted_filter.each do |filter_field, filter_value|
+      notes = notes.where(filter_field => filter_value)
+    end
+
+    notes
+  }
+
+  scope :paginated, lambda { |page, page_size|
+    notes = all
+    notes = notes.page(page).per(page_size)
+
+    notes
+  }
```

Comment:
No solemos utilizar asignaciones de variables de este modo, y en este contexto del scope no es necesario hacer la referencia `all`, ni tampoco hacer un retorno explícito al final
Entiendo que con algo así debería funcionar:
```suggestion
  scope :paginated, lambda { |page, page_size| page(page).per(page_size) }
```